Tools/test/IntegrationTest.cpp

```diff
@@ -0,0 +1,249 @@
+            <DecayProducts>
+              <Particle Name='gamma' FinalState='0' Helicity='1' />
+              <Particle Name='f0' FinalState='1' Helicity='0' />
+            </DecayProducts>
+            <CanonicalSum L='0' S='1.0'>
+              <ClebschGordan Type='LS' j1='0.0' m1='0.0' j2='1.0' m2='1.0' J='1.0' M='1.0' />
+              <ClebschGordan Type='s2s3' j1='1.0' m1='1.0' j2='0.0' m2='0.0' J='1.0' M='1.0' />
+            </CanonicalSum>
+          </Amplitude>
+        </Amplitude>
+        <Parameter Class='Double' Type='Magnitude' Name='Magnitude_jpsi_to_gamma+f0_L_0.0_S_1.0;'>
+          <Value>1.0</Value>
+          <Fix>false</Fix>
+        </Parameter>
+        <Parameter Class='Double' Type='Phase' Name='Phase_jpsi_to_gamma+f0_L_0.0_S_1.0;'>
+          <Value>0.0</Value>
+          <Fix>false</Fix>
+        </Parameter>
+      </Amplitude>
+
+      <Amplitude Class='CoefficientAmplitude' Name='jpsi_1_to_gamma_1+f0_0_L_2.0_S_1.0;'>
+        <Amplitude Class='SequentialAmplitude' Name='jpsi_1_to_gamma_1+f0_0_L_2.0_S_1.0;'>
+          <Amplitude Class='HelicityDecay' Name='jpsi_to_gamma_f0_L2_S1'>
+            <DecayParticle Name='jpsi' Helicity='+1' />
+            <DecayProducts>
+              <Particle Name='gamma' FinalState='0' Helicity='1' />
+              <Particle Name='f0' FinalState='1' Helicity='0' />
+            </DecayProducts>
+            <CanonicalSum L='2' S='1.0'>
+              <ClebschGordan Type='LS' j1='2.0' m1='0.0' j2='1.0' m2='1.0' J='1.0' M='1.0' />
+              <ClebschGordan Type='s2s3' j1='1.0' m1='1.0' j2='0.0' m2='0.0' J='1.0' M='1.0' />
+            </CanonicalSum>
+          </Amplitude>
+        </Amplitude>
+        <Parameter Class='Double' Type='Magnitude' Name='Magnitude_jpsi_to_gamma+f0_L_2.0_S_1.0;'>
+          <Value>1.0</Value>
+          <Fix>false</Fix>
+        </Parameter>
+        <Parameter Class='Double' Type='Phase' Name='Phase_jpsi_to_gamma+f0_L_2.0_S_1.0;'>
+          <Value>0.0</Value>
+          <Fix>false</Fix>
+        </Parameter>
+      </Amplitude>
+    </Intensity>
+)####";
+
+BOOST_AUTO_TEST_CASE(IntegrationTest) {
+  ComPWA::Logging Log("trace", "");
+  LOG(INFO) << "Now check saven and load fit result...";
+
+  boost::property_tree::ptree ModelTree;
+  std::stringstream ModelStream;
+  ComPWA::Physics::IntensityBuilderXML Builder;
+
+  // Particle list
+  ModelStream << TestParticles;
+  boost::property_tree::xml_parser::read_xml(ModelStream, ModelTree);
+  auto PartL = std::make_shared<ComPWA::PartList>();
+  ReadParticles(PartL, ModelTree);
+
+  // Kinematics
+  ModelStream.clear();
+  ModelTree = boost::property_tree::ptree();
+  ModelStream << JpsiDecayKinematics;
+  boost::property_tree::xml_parser::read_xml(ModelStream, ModelTree);
+  auto DecayKin = Builder.createHelicityKinematics(
+      PartL, ModelTree.get_child("HelicityKinematics"));
+
+  // Estimated using 10^7 events: 0.198984549+-5.627720404e-05
+  double TrueIntegral = 0.198984549;
```

Comment:
How did you calculate the true integral?

Reply:
Its estimated using a high number of phsp events. I hardcode it here to save memory on TravisCI and also to indicate if the model changes

Reply:
Ok, thats fine. I think it would be not bad if we would also include a test for the correctness of the integration (accuracy test). So maybe integrate a gaussian or a sphere.